main/src/observables/gravitational_waves.hpp

```diff
@@ -0,0 +1,125 @@
+/*
+* MIT License
+*
+* Copyright (c) 2021 CSCS, ETH Zurich
+*               2021 University of Basel
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+/*! @file output time, energies and the gravitational radiation as well as the quadrupole moment
+ *
+ *  @author Lukas Schmidt
+ */
+
+#include <array>
+#include "mpi.h"
+#include "iobservables.hpp"
+#include "io/ifile_writer.hpp"
+#include "grav_waves_calculations.hpp"
+
+namespace sphexa
+{
+
+/*!@brief calculates the gravitational waves observable
+ *
+ * @tparam T            floating point type
+ * @tparam Dataset
+ * @param d             Dataset
+ * @param first         first locally assigned particle index of buffers in @p d
+ * @param last          first locally assigned particle index of buffers in @p d
+ * @param viewTheta     viewing angle to determine ...
+ * @param viewPhi       viewing angle to determine ...
+ * @return              array containing the observables and the second derivative of the quadpole momentum:
+ *                      {httplus, httcross, ixx, iyy, izz, ixy, ixz, iyz}
+ */
+template<class T, class Dataset>
+std::array<T, 8> gravRad(Dataset& d,size_t first, size_t last, T viewTheta, T viewPhi)
+{
+
+    std::array<T, 6> d2Qxx_local;
+
+    d2Qxx_local[0] = d2QuadpoleMomentum<T>(first, last, 0, 0,
```

Comment:
These 6 calls now become
```c++
d2Q_local[Q::xx] = d2QuadpoleMomentum<T>(first, last, Dim::x, Dim::x,
```
etc

Reply:
This should address Rubens suggestion about giving understandable names, while
keeping the 6 values in an array which is needed for the global reduction.